libtenzir/src/pipeline_executor.cpp

```diff
@@ -62,10 +62,7 @@ void pipeline_executor_state::start_nodes_if_all_spawned() {
         finish_start();
       },
       [this](const caf::error& err) mutable {
-        if (err == caf::sec::broken_promise) {
-          abort_start(ec::silent);
-          return;
-        }
+        TENZIR_ASSERT(err != caf::sec::broken_promise);
```

Comment:
That's too harsh—what if an operator exits with a broken promise?

Reply:
Agreed.

FWIW: I still get the `broken promise` diagnostic in the pipeline overview with this branch, the actual error is printed on stderr:
```
*** unhandled exception: [id: 362, name: user.scheduled-actor, exception typeid: boost.wrapexcept]: bind: Address already in use [system:98 at include/boost/asio/detail/reactive_socket_service.hpp:161:33 in function 'boost::system::error_code boost::asio::detail::reactive_socket_service<boost::asio::ip::tcp>::bind(implementation_type &, const endpoint_type &, boost::system::error_code &) [Protocol = boost::asio::ip::tcp]']
```
Pretty ugly, but at least it contains "Address already in use".

Reply:
Hm, I think this is a local problem in `tcp-listen`.

Reply:
I'll create a separate PR to generally catch exceptions in operators to turn them into diagnostics.

Reply:
I removed the assetion.

Reply:
Great, think this can be merged then. I'll open a follow-up PR shortly.